tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+do! Let's first download the tarball:
+
+```bash
+wget https://dlcdn.apache.org/httpd/httpd-2.4.62.tar.gz
+```
+
+#### Verifying the checksums
+
+For this particular version of `apache` there is no MD5 checksum file so we
+can't use the `md5sum` tool we used earlier. But, there *are* SHA256 and SHA512
+checksum files. Let's download the SHA256 checksum file -- it's advised to
+download these from the official source, so let's do that:
+
+```bash
+wget https://downloads.apache.org/httpd/httpd-2.4.62.tar.gz.sha256
+```
+
+Then we can verify the package download against that checksum using the
+`sha256sum` tool:
+
+```bash
+sha256sum -c httpd-2.4.62.tar.gz.sha256
+```
+
+Which should return:
+
+```bash
+httpd-2.4.62.tar.gz: OK
+```
+
+This tells us that the package file is validated. If the output returned
+`FAILED` then we would know something is wrong with it. In that case, we would
+want to download the file again (from a trusted source, preferably). We
+definitely wouldn't want to install it if the check fails! 
+
+In our case, since we obtained the package from a trusted source (the official
+website) and the checksums matched, we can proceed with the installation.
+
+#### Unpacking and installing
+
+First,  let's "un-tar" the tarball:
+
+```bash
+tar -xvzf httpd-2.4.62.tar.gz
+```
+
+This will unpack the contents of the tarball into a directory of the same name,
+which we can change to:
+
+```bash
+cd httpd-2.4.62
+```
+
+If we run `ls --all` we can see a list of all the files that are contained in
+the package.
+
+There are a couple of dependencies that we'll want to install that don't come
+packaged with the development version, so let's install those now:
+
+```bash
+sudo apt install autoconf libtool-bin
+```
+
+Next we'll want to configure the source tree -- the prefix is the most
+important option, so let's set that (for the sake of this example, we'll just
+set it to the default location):
+
+```bash
+./configure --prefix=/usr/local/apache2
+```
+
+Configuration is an important step, because this is where the Makefiles are
+created that will allow us to install the package. It will take a few moments
+as it checks all the files.
+
+Next, we want to compile by running `make`, and then install by running
+`make install`:
+
+```bash
+make
+make install
+```
+
+Now at this point, we could consider ourselves done, but installing this way
+means that there's no easy way to uninstall or update the package in the
+future since the package managers (`apt` and `dpkg`) can't track them. So one
+thing we can do is to create an installable deb package (which will help to
+make removal easier!) -- to do that, we can use the `fpm` utility.
+This is a Ruby package that needs to be installed via `gem`:
+
+```bash
+sudo apt install ruby ruby-dev gcc
+gem install fpm
+```
+
+Then we can create the package:
+
+```bash
+fpm -s dir -t deb -n apache2-custom -v 2.4.62 /usr/local/apache2
```

Comment:
can you give context to what’s in this command? (say what all the flags are/etc).

I also got this error:
```bash
Invalid package configuration: Cannot package the path './usr/local/apache2', does it exist? {:level=>:error}
```

Reply:
Context added!